engine/runtime/src/main/java/org/enso/interpreter/runtime/state/HasContextEnabledNode.java

```diff
@@ -0,0 +1,50 @@
+package org.enso.interpreter.runtime.state;
+
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.atom.AtomConstructor;
+
+/**
+ * A node representing the functionality done by {@code Standard.Base.Runtime.Context.is_enabled}.
+ */
+@GenerateUncached
+public abstract class HasContextEnabledNode extends Node {
+
+  public static HasContextEnabledNode getUncached() {
+    return HasContextEnabledNodeGen.getUncached();
+  }
+
+  public static HasContextEnabledNode create() {
+    return HasContextEnabledNodeGen.create();
+  }
+
+  /**
+   * Returns true if the context represented by the given {@code runtimeCtxCtor} is enabled in the
+   * given {@code executionEnvironment}.
+   *
+   * @param runtimeCtxCtor Constructor of {@code Standard.Base.Runtime.Context}.
+   */
+  public abstract boolean executeHasContextEnabled(
+      ExecutionEnvironment executionEnvironment, AtomConstructor runtimeCtxCtor);
+
+  @Specialization(guards = "executionEnvironment == cachedEnv", limit = "3")
+  boolean doIt(
+      ExecutionEnvironment executionEnvironment,
+      AtomConstructor runtimeCtxCtor,
+      @Cached("executionEnvironment") ExecutionEnvironment cachedEnv) {
```

Comment:
With cached `ExecutionEnvironment`, the performance improved from **1.179 ± 0.648 to 0.218 ± 0.014**.